src/Dhall/Parser/Expression.hs

```diff
@@ -171,6 +171,14 @@ makeOperatorExpression subExpression operatorParser operator embedded =
         b <- many (do operatorParser; subExpression embedded)
         return (foldr1 operator (a:b)) )
 
+importAltExpression :: Parser a -> Parser (Expr Src a)
+importAltExpression embedded =
+  noted (do
+    l <- orExpression embedded
+    a <- embedded  -- FIXME: broken
```

Comment:
You can just do:

```haskell
let a = Missing
```

Reply:
Not quite, since we need to keep the parsers polymorphic (as it gets concrete only in `exprA = expr import_`).
I think the problem here is that the constructor is `ImportAlt a (Expr s a) (Expr s a)`, and I'm not sure how to get an `a` from the `embedded` without running it (which we cannot, as we have to continue the parsing of the current tree), so that I can run the `fold` two lines later with `(ImportAlt a)`. That's why I'm concerned that the constructor might not be right.